fili-core/src/test/groovy/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProviderSpec.groovy

```diff
@@ -271,4 +196,26 @@ class LuceneSearchProviderSpec extends SearchProviderSpec<LuceneSearchProvider>
                 throw it.cause
             } }
     }
+
+    /**
+     * Generate and returns a list of Lucene index files under a specified path.
+     *
+     * @param path  The specified path under which the index files are generated
+     *
+     * @return the list of Lucene index files under the specified path
+     */
+    List<Path> generateIndexFiles(Path path) {
```

Comment:
In order to reopen Lucene index after index files are swapped, new index files must be a valid **non-empty** index files(exception thrown on empty index files). This method creates the real index files